canvas_modules/common-canvas/__tests__/common-canvas/common-canvas-test.js

```diff
@@ -31,54 +33,82 @@ describe("CommonCanvas renders correctly", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, { showBottomPanel: false }, {});
+		expect(wrapper.find(CanvasBottomPanel)).to.have.length(1);
+		expect(canvasController.isBottomPanelOpen() === false).to.be.true;
+	});
+
+	it("should render one <CommonCanvasRightFlyout/> component when showRightFlyout is true", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, {}, { showRightFlyout: false });
+		expect(wrapper.find(CommonCanvasRightFlyout)).to.have.length(1);
+		expect(canvasController.isRightFlyoutOpen() === true).to.be.false;
+	});
+
+	it("should render one <CommonCanvasRightFlyout/> component when showRightFlyout is false", () => {
+		const config = {};
+		const wrapper = createCommonCanvas(config, canvasController, {}, { showRightFlyout: true });
+		expect(wrapper.find(CommonCanvasRightFlyout)).to.have.length(1);
+		expect(canvasController.isRightFlyoutOpen() === false).to.be.false;
+	});
+
 	it("should render one <CanvasContents/> component", () => {
 		const config = {};
-		const wrapper = createCommonCanvas(config, canvasController);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {});
 		expect(wrapper.find(CanvasContents)).to.have.length(1);
 	});
 
 	it("should render one <PaletteDialog/> component when enablePaletteLayout is set to Modal", () => {
 		const config = { enablePaletteLayout: "Modal" };
-		const wrapper = createCommonCanvas(config, canvasController);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {});
 		expect(wrapper.find(PaletteDialog)).to.have.length(1);
 	});
 
 	it("should render one <PaletteFlyout/> component when enablePaletteLayout is not specified", () => {
 		const config = {};
-		const wrapper = createCommonCanvas(config, canvasController);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {});
 		expect(wrapper.find(PaletteFlyout)).to.have.length(1);
 	});
 
 	it("should render one <PaletteFlyout/> component when enablePaletteLayout is set to Flyout", () => {
 		const config = { enablePaletteLayout: "Flyout" };
-		const wrapper = createCommonCanvas(config, canvasController);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {});
 		expect(wrapper.find(PaletteFlyout)).to.have.length(1);
 	});
 
 	it("should not render any <PaletteDialog/> component when enablePaletteLayout is not specified", () => {
 		const config = {};
-		const wrapper = createCommonCanvas(config, canvasController);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {});
 		expect(wrapper.find(PaletteDialog)).to.have.length(0);
 	});
 
 	it("should render one <Toolbar/> component when toolbarConfig is provided", () => {
 		const toolbarConfig = [];
 		const config = {};
-		const wrapper = createCommonCanvas(config, canvasController, toolbarConfig);
+		const wrapper = createCommonCanvas(config, canvasController, {}, {}, toolbarConfig);
```

Comment:
In the createCommonCanvas function,  if you put the `canvasParams` parameter at the end of the set of parameters AND if the function is able to handle the situation where `canvasParams` is undefined (as mentioned in the comment above) then you wouldn't have to specify a `{}` here. This would apply to most of the changes below nor any of the places below.